Sources/SwiftParser/Statements.swift

```diff
@@ -570,6 +570,16 @@ extension Parser {
     let tryKeyword = self.consume(if: .keyword(.try))
     let awaitKeyword = self.consume(if: .keyword(.await))
 
+    let unsafeKeyword: RawTokenSyntax?
+    if let modifierKeyword = ExpressionModifierKeyword(
+      lexeme: self.currentToken,
+      experimentalFeatures: self.experimentalFeatures
+    ), modifierKeyword == .unsafe, !self.peek(isAt: .keyword(.in)) {
+      unsafeKeyword = self.consumeAnyToken(remapping: .keyword)
```

Comment:
I think we also need to check at least for `:` to make sure the following continues to parse:

```swift
for unsafe: Int in [1, 2] {
  print(unsafe)
}
```

Also, I’d prefer to do `self.expect(.keyword(.unsafe))` instead of `consumeAnyToken`. It just adds more context to what kind of keyword `unsafeKeyword` is and another layer of safety if the current token should not be `unsafe` for any reason in the future.

Reply:
Hah, I read your comments out of order. We indeed need to deal with the `:`. Thanks for the `self.expect(.keyword(.unsafe))` hint.